lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -78,10 +82,15 @@ def get_vars(self, loader, path, entities, cache=True):
 
         data = {}
         for entity in entities:
-            if isinstance(entity, Host):
+            entity_id = id(entity)
+            if entity_id in HOSTS or entity_id in GROUPS:
```

Comment:
'basedir' is mostly 'ported over' from when vars plugins were hardcoded plugins inside inventory and vars manager code. I always meant to clean it up, but was never a priority.

Reply:
internally we use host name and group name as unique identifiers everywhere else, we should keep that here as the host/group objects can be regenerated and have diff ids even for same host/group.

Reply:
Since we've always been requiring specific types here anyway with the `isinstance`, and we fully control those types, we can actually do this a lot cheaper by either just doing `type(entity) is Host`, or if we want to allow for the possibility of subclasses, do something silly like add new class-level attrs on `Host` and `Group` like `base_type = Host`, then still the runtime cost is just `entity.base_type is Host` (makes the eval cost even cheaper at just an attr lookup and a reference compare).

Reply:
I added  the new class attr in an abundance of caution and removed the `isinstance` calls.